app/validators/answers/number_answer_validator.py

```diff
@@ -65,11 +66,39 @@ def validate_value_in_limits(self):
             self.add_error(
                 self.MINIMUM_LESS_THAN_LIMIT, value=min_value, limit=MIN_NUMBER
             )
+        elif isinstance(min_value, dict):
+            source = min_value["source"]
+            answer_ranges = self.questionnaire_schema.numeric_answer_ranges
+            referred_answer = (
+                self.questionnaire_schema.get_numeric_value_for_value_source(
+                    source, min_value, answer_ranges
+                )
+            )
+            if referred_answer["min"] < MIN_NUMBER:
+                self.add_error(
+                    self.MINIMUM_LESS_THAN_LIMIT,
+                    value=referred_answer["min"],
+                    limit=MIN_NUMBER,
+                )
 
         if isinstance(max_value, int) and max_value > MAX_NUMBER:
             self.add_error(
                 self.MAXIMUM_GREATER_THAN_LIMIT, value=max_value, limit=MAX_NUMBER
             )
+        elif isinstance(min_value, dict):
+            source = min_value["source"]
+            answer_ranges = self.questionnaire_schema.numeric_answer_ranges
+            referred_answer = (
+                self.questionnaire_schema.get_numeric_value_for_value_source(
+                    source, min_value, answer_ranges
+                )
+            )
+            if referred_answer["max"] > MAX_NUMBER:
+                self.add_error(
```

Comment:
I think you meant for this to be max_value, currently its min_value twice

Reply:
Changed max_value 👍 on line 99